### PR TITLE
feat(dialog): add prop action

### PR DESCRIPTION
**Pull request by bljessica**

### CODE REVIEW

File: src/dialog/demos/enUS/index.demo-entry.md [hunk 1]
```diff
@@ -58,6 +59,7 @@ use-component
 | bordered | `boolean` | `false` | Whether to show border. |
 | closable | `boolean` | `true` | Whether to show close icon. |
 | content | `string \| (() => VNodeChild)` | `undefined` | Content, can be a render function. |
+| action | `() => VNodeChild` | `undefined` | Content of the operation area, needs to be a render function. |
```
Comment (Talljack) [inline src/dialog/demos/enUS/index.demo-entry.md:62]:
```suggestion
| action | `() => VNodeChild` | `undefined` | Content of the operation area, must be a render function. |
```

File: src/dialog/demos/enUS/action-slot.demo.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# 基础用法
+
```
Comment (Talljack) [inline src/dialog/demos/enUS/action-slot.demo.md:2]:
```suggestion
# Custom action
```

File: src/dialog/demos/zhCN/action-slot.demo.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# 基础用法
```
Comment (Talljack) [inline src/dialog/demos/zhCN/action-slot.demo.md:1]:
```suggestion
# 自定义 action
```

**Comment by Talljack** [comment]
Please add test cases

**Review by Volankey — CHANGES_REQUESTED** [review]

File: src/dialog/src/Dialog.tsx [hunk 4]
```diff
@@ -243,40 +245,48 @@ export default defineComponent({
         <div class={`${mergedClsPrefix}-dialog__content`}>
           {renderSlot($slots, 'default', undefined, () => [render(content)])}
         </div>
-        {$slots.action || (!$slots.action && (positiveText || negativeText)) ? (
+        {$slots.action ||
+        (!$slots.action && (positiveText || negativeText || action)) ? (
```
Comment (07akioni) [inline src/dialog/src/Dialog.tsx:249]:
```suggestion
        {($slots.action || positiveText || negativeText || action) ? (
```

File: src/dialog/src/Dialog.tsx [hunk 5]
```diff
@@ -243,40 +245,48 @@ export default defineComponent({
...
+                        {{
+                          default: () => render(this.positiveText)
+                        }}
+                      </NButton>
+                    )
+                  ]
+            )}
           </div>
-        ) : null}
+            ) : null}
```
Comment (07akioni) [inline src/dialog/src/Dialog.tsx:289]:
```suggestion
        ) : null}
```

**Pull request merged**